sdcm/cluster_gce.py

```diff
@@ -123,8 +124,15 @@ def reboot(self, hard=True, verify_ssh=True):
         pre_uptime = result.stdout
 
         def uptime_changed():
-            result = self.remoter.run('uptime -s', ignore_status=True)
-            return pre_uptime != result.stdout
+            try:
+                result = self.remoter.run('uptime -s', ignore_status=True)
+                return pre_uptime != result.stdout
+            except SSHException as ex:
+                self.log.debug("Network isn't available, reboot might already start, %s" % ex)
+                return True
```

Comment:
I think we should assume not, and pass False

Reply:
Sometimes, we want to do something after the node (scylla-server) isn't available, so the it's optional to wait the ssh is up in reboot(hard=True, verify_ssh=True). If we wait until we really get result by ssh, node reboot completes (network is up again), then we lost chance to do sometime after reboot started.

It has less chance that node isn't reboot but the network has problem and raises SSHException, and network recovers later, but reboot still not starts. Then it's a big problem.

I just rechecked the code, we didn't have code to disable verify_ssh in calling reboot().
So it's fine to return False in this case.

Reply:
> Sometimes, we want to do something after the node (scylla-server) isn't available, so the it's optional to wait the ssh is up in reboot(hard=True, verify_ssh=True). If we wait until we really get result by ssh, node reboot completes (network is up again), then we lost chance to do sometime after reboot started.
> 
> It has less chance that node isn't reboot but the network has problem and raises SSHException, and network recovers later, but reboot still not starts. Then it's a big problem.
> 
> I just rechecked the code, we didn't have code to disable verify_ssh in calling reboot().
> So it's fine to return False in this case.

In this case, we need to increase wait timeout to be same as wait_ssh_up(), timeout == 500 seconds.